src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat.slice(-1));
+  const [format, separator] = [toFormat.slice(0, -1), toFormat.slice(-1)];
+  const temp = {};
+  const resultDate = [];
+
+  // making object to save all data from dateArray
+  for (let i = 0; i < 3; i++) {
+    temp[fromFormat[i][0]] = dateArray[i];
+  };
+
+  // formatting year
+  switch (temp.Y.length) {
+    case 4 :
+      temp.Y = toFormat.indexOf('YY') === -1
+        ? temp.Y
+        : temp.Y.slice(-2);
+      break;
+
+    default :
+      temp.Y = toFormat.indexOf('YY') === -1
+        ? +temp.Y < 30
```

Comment:
use `Number` instead `+`
```suggestion
        ? Number(temp.Y < 30)
```